### PR TITLE
Add goldpinger

### PR DESCRIPTION
**Pull request by phillebaba**
This change adds goldpinger with a servicemonitor to scrape its metrics.

Closes #269

### CODE REVIEW

File: modules/kubernetes/aks-core/modules.tf [hunk 1]
```diff
@@ -372,3 +373,15 @@ module "xenit" {
   loki_api_fqdn        = var.xenit_config.loki_api
 }
 
+module "goldpinger" {
+  depends_on = [module.opa_gatekeeper, module.linkerd]
```
Comment (NissesSenap) [inline modules/kubernetes/aks-core/modules.tf:377]: Do we need linkerd to run goldpinger? Can't we use this on all our clusters for network monitoring?

File: modules/kubernetes/aks-core/modules.tf [hunk 2]
```diff
@@ -372,3 +373,15 @@ module "xenit" {
   loki_api_fqdn        = var.xenit_config.loki_api
 }
 
+module "goldpinger" {
```
Comment (simongottschlag) [inline modules/kubernetes/aks-core/modules.tf:376]: Make linkerd optional

File: modules/kubernetes/goldpinger/templates/values.yaml.tpl [hunk 3]
```diff
@@ -0,0 +1,7 @@
+goldpinger:
+  port: 8080
+
+podAnnotations:
+  linkerd.io/inject: enabled
```
Comment (simongottschlag) [inline modules/kubernetes/goldpinger/templates/values.yaml.tpl:8]: make this optional and based on the linkerd_enabled var

File: modules/kubernetes/prometheus/charts/prometheus-extras/values.yaml [hunk 4]
```diff
@@ -41,3 +41,4 @@ prometheus:
 
 enabledMonitors:
   linkerd: false
+  goldpinger: false
```
Comment (simongottschlag) [inline modules/kubernetes/prometheus/charts/prometheus-extras/values.yaml:44]: enabled by default?

File: modules/kubernetes/goldpinger/templates/values.yaml.tpl (hunk 3 above)
Comment (NissesSenap) [inline modules/kubernetes/goldpinger/templates/values.yaml.tpl:8]: Add a if statment check if linkerd enabled or not

File: modules/kubernetes/aks-core/modules.tf (hunk 1 above)
Comment (NissesSenap) [reply modules/kubernetes/aks-core/modules.tf:377]: Is terrafrom smart enough that if linkerd module is disabled it will still continue?

**Review by NissesSenap — COMMENTED** [review]
Minor comments

File: modules/kubernetes/aks-core/modules.tf (hunk 1 above)
Comment (phillebaba) [reply modules/kubernetes/aks-core/modules.tf:377]: Yes terraform will be smart enough to ignore this if linkerd is not enabled.

File: modules/kubernetes/goldpinger/templates/values.yaml.tpl (hunk 3 above)
Comment (phillebaba) [reply modules/kubernetes/goldpinger/templates/values.yaml.tpl:8]: Done

Comment (phillebaba) [reply modules/kubernetes/goldpinger/templates/values.yaml.tpl:8]: Done

File: modules/kubernetes/aks-core/modules.tf (hunk 2 above)
Comment (phillebaba) [reply modules/kubernetes/aks-core/modules.tf:376]: Done

**Review by simongottschlag — APPROVED** [review]
Great work @phillebaba ! :100:

**Pull request merged**